lib/list.fz

```diff
@@ -231,6 +231,34 @@ list(A type) : choice nil (Cons A (list A)), Sequence A is
     "{(map string x->x.asString).fold (strings.concat sep)}"
 
 
+  # add an element sep in front of every element of this list.
+  #
+  prepend_to_all(sep A) list A is
+    prepend_to_all0 sep nil
+
+
+  # add an element sep in front of every element of this list, helper to
+  # allow tail recursion.
+  #
+  # if this list is the empty list, return the given result list res, if
+  # not, make a recursively call this on the tail of this list, feeding it
```

Comment:
_if not_ -> _otherwise_
I cannot parse _make a recursively call this on the tail of this list_